src/tests/ftest/launch.py

```diff
@@ -1163,6 +1165,21 @@ def rename_logs(avocado_logs_dir, test_file):
             "Error renaming {} to {}: {}".format(
                 test_logs_dir, new_test_logs_dir, error))
 
+    try:
+        xml_file = os.path.join(new_test_logs_dir, "results.xml")
+        with open(xml_file) as xml_buffer:
+            xml_data = xml_buffer.read()
+
+        file_parts = os.path.split(test_file)
+        c_name = os.path.splitext(os.path.basename(file_parts[0]))[0]
+        org_class = "classname=\""
+        new_class = "classname=\"FTEST_{}.".format(c_name)
+        xml_data = re.sub(org_class, new_class, xml_data)
+
+        with open(xml_file, "w") as xml_buffer:
+            xml_buffer.write(xml_data)
+    except OSError as error:
+        print("Error updating classname in {}".format(xml_file))
```

Comment:
A try/except clause should really cover the minimum amount of code to, in general, avoid masking bugs.  It also has the benefit of being able to provide a more detailed reason for the error.  

In this particular situation the OSError exception handling does not appear to necessarily mask any issues as the non `open()` commands should not be raising that particular exception, but it would be nice to indicate more specifically whether the error was with the read or the write.  In addition we should log the exception being raised, in this case `error`, so we know why we hit the failure.

We can also simplify the extraction of the directory from the functional test.

For example:
```suggestion
    xml_file = os.path.join(new_test_logs_dir, "results.xml")
    try:
        with open(xml_file) as xml_buffer:
            xml_data = xml_buffer.read()
    except OSError as error:
        print(
            "Error updating xml classname for {}, reading {} : {}".format(
            test_name, xml_file, str(error)))
        return
        
    test_dir = os.path.split(os.path.dirname(test_file))[-1]
    org_class = "classname=\""
    new_class = "classname=\"FTEST_{}.".format(test_dir)
    xml_data = re.sub(org_class, new_class, xml_data)

    try:
        with open(xml_file, "w") as xml_buffer:
            xml_buffer.write(xml_data)
    except OSError as error:
        print(
            "Error updating xml classname for {}, writing {}: {}".format(
            test_name, xml_file, str(error)))
```